### PR TITLE
created company reviews stats section

### PR DESCRIPTION
**Pull request by illyadmytrenko**

### CODE REVIEW

File: src/common/constants/reviews-sort-options.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const ReviewsSortOptions = [
+	{
+		label: "Найпопулярніші",
+		value: "avg_overall_rating",
+	},
+	{
+		label: "Найнадійніші",
+		value: "avg_overall_rating",
+	},
```
[the remark this answers is not in the document]
Comment (illyadmytrenko) [reply src/common/constants/reviews-sort-options.ts:9]: i couldn't find them too, so I just made them as the example.

File: src/common/types/company/company.type.ts [hunk 2]
```diff
@@ -12,6 +13,7 @@ type Company = {
 	logo: string;
 	name: string;
 	phone_numbers: string[];
+	review_count: number;
 	total_courses: number;
 	total_reviews_count: number;
```
[the remark this answers is not in the document]
Comment (illyadmytrenko) [reply src/common/types/company/company.type.ts:18]: As far as I understand, review_count is count of reviews for only a company, the same goes for avg_rating, but total_reviews_count is count of reviews not only for company but also for courses that this company provide, so it's simply a sum of the sum of reviews of everything related to the company. And the same goes to avg_overall_rating.

[the remark this answers is not in the document]
Comment (ValentinaTenus) [reply src/common/types/company/company.type.ts:18]: thanks a lot

File: src/pages/company-details-page/components/company-details/components/contacts/contacts.tsx [hunk 3]
```diff
@@ -32,12 +38,37 @@ const Contacts = React.forwardRef<HTMLDivElement, { company: Company }>(
...
+								>
+									{phone_number}
+								</span>
+							))}
+						</li>
 					</ul>
 					<div className={styles["contacts_buttons"]}>
-						<Button variant={ButtonVariant.PRIMARY}>
+						<Button
+							onClick={() => (window.location.href = `mailto:${company.email}`)}
```
Comment (ValentinaTenus) [inline src/pages/company-details-page/components/company-details/components/contacts/contacts.tsx:63]: I'd create handler for this

File: src/pages/company-details-page/components/company-details/components/reviews/components/components/review-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	const MIN_TEXT = 200;
+
+	const [rating, setRating] = useState<null | number>(ZERO);
+	const [reviewText, setReviewText] = useState("");
+
+	const [sendReview] = useSendReviewMutation();
+
+	const handleReviewChange = (
+		event: React.ChangeEvent<HTMLTextAreaElement>,
+	) => {
```
Comment (ValentinaTenus) [inline src/pages/company-details-page/components/company-details/components/reviews/components/components/review-modal.tsx:26]: Use useCallback hook

File: src/pages/company-details-page/components/company-details/components/reviews/components/components/review-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
+		const response = await sendReview({
+			companyId: 1,
+			rating: rating,
+			text: reviewText,
+		});
+		onClose();
+	};
+
+	const handleCloseReviewModal = () => {
```
Comment (ValentinaTenus) [inline src/pages/company-details-page/components/company-details/components/reviews/components/components/review-modal.tsx:40]: Use useCallback for handlers

**Review by ValentinaTenus — APPROVED** [review]

File: src/common/components/modal/styles.module.scss [hunk 6]
```diff
@@ -25,7 +25,7 @@
 
 .modal__header {
 	align-items: center;
-	border-bottom: 1px solid var(--color-grey-300);
+	// border-bottom: 1px solid var(--color-grey-300);
```
Comment (ValentinaTenus) [inline src/common/components/modal/styles.module.scss:28]: delete comment

File: src/pages/company-details-page/components/company-details/components/reviews/components/components/report-modal.tsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+		[setReportText],
+	);
+
+	const handleCloseReportModal = useCallback(() => {
+		onClose();
+		setReportText("");
+	}, [onClose, setReportText]);
+
+	const handleSubmit = useCallback(async () => {
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (ValentinaTenus) [inline src/pages/company-details-page/components/company-details/components/reviews/components/components/report-modal.tsx:39]: why do you need this

File: src/pages/company-details-page/components/company-details/components/reviews/components/components/review-modal.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+	}, [onClose, setRating, setReviewText]);
+
+	const handleSubmit = useCallback(async () => {
+		const response = await sendReview({
+			companyId: company.id,
+			rating: rating,
+			text: reviewText,
+		});
+		// eslint-disable-next-line no-console
+		console.log(response);
```
Comment (ValentinaTenus) [inline src/pages/company-details-page/components/company-details/components/reviews/components/components/review-modal.tsx:46]: delete

File: src/pages/company-details-page/components/company-details/components/reviews/components/components/share-modal.tsx [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+	) => {
+		if (shareableLink) {
+			const encodedLink = encodeURIComponent(shareableLink);
+			switch (platform) {
+				case "facebook":
+					return `https://www.facebook.com/sharer/sharer.php?u=${encodedLink}`;
+				case "linkedin":
+					return `https://www.linkedin.com/sharing/share-offsite/?url=${encodedLink}`;
+				case "x":
+					return `https://twitter.com/intent/tweet?url=${encodedLink}`;
```
Comment (ValentinaTenus) [inline src/pages/company-details-page/components/company-details/components/reviews/components/components/share-modal.tsx:46]: add to env and env.example

**Review by ValentinaTenus — CHANGES_REQUESTED** [review]

**Pull request merged**